bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/connectivity/LocalHttpConnector.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal.connectivity;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LocalHttpConnector} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class LocalHttpConnector implements SolaxConnector {
+
+    private static final int HTTP_REQUEST_TIME_OUT = 5000;
+
+    private static final String CONTENT_TYPE = "text/html; charset=utf-8";
+
+    private final Logger logger = LoggerFactory.getLogger(LocalHttpConnector.class);
+
+    private static final String OPT_TYPE = "optType";
+    private static final String READ_REALTIME_DATA = "ReadRealTimeData";
+
+    private static final String PASSWORD = "pwd";
+    // The serial number of the Wifi dongle is the password for the connection (default)
+    private String passwordValue;
+    private String uri;
+
+    public LocalHttpConnector(String passwordValue, String host) {
+        this.passwordValue = passwordValue;
+        this.uri = "http://" + host;
+    }
+
+    @Override
+    public @Nullable String retrieveData() throws IOException {
```

Comment:
The `SolaxInterface` declares throwing `Exception`, but I think `IOException` is sufficient, so probably should be changed in the interface.